hs_restclient/__init__.py

```diff
@@ -1 +1,666 @@
+            if type(to_date) is datetime.date:
+                params['to_date'] = to_date.strftime('%Y-%m-%d')
+            else:
+                raise HydroShareArgumentException("to_date must of type '{0}'.".format(datetime.date))
+        if types:
+            if not is_sequence(types):
+                raise HydroShareArgumentException("Types must be a sequence type, but not a string.")
+            params['types'] = types
+
+        num_resources = 0
+
+        # Get first (only?) page of results
+        r = self._request('GET', url, params=params)
+        if r.status_code != 200:
+            raise HydroShareHTTPException((url, 'GET', r.status_code, params))
+        res = r.json()
+        tot_resources = res['count']
+        resources = res['results']
+        num_resources += len(res['results'])
+
+        for r in resources:
+            yield r
+
+        # Get remaining pages (if any exist)
+        while res['next'] and num_resources < tot_resources:
+            r = self._request('GET', res['next'], params=params)
+            if r.status_code != 200:
+                raise HydroShareHTTPException((url, 'GET', r.status_code, params))
+            res = r.json()
+            resources = res['results']
+            for r in resources:
+                yield r
+            num_resources += len(res['results'])
+
+        if num_resources != tot_resources:
+            raise HydroShareException("Expected {tot} resources but found {num}.".format(tot_resources, num_resources))
+
+    def getSystemMetadata(self, pid):
+        """ Get system metadata for a resource
+
+        :param pid: The HydroShare ID of the resource
+
+        :raises: HydroShareHTTPException to signal an HTTP error
+
+        :return: A dict representing the JSON object representation of the resource returned by the REST end point.
+
+        Example of data returned:
+
+        {u'bag_url': u'http://www.hydroshare.org/static/media/bags/hr3hy35y5ht4y54hhthrtg43w.zip',
+          u'creator': u'B Miles',
+          u'date_created': u'01-02-2015',
+          u'date_last_updated': u'05-13-2015',
+          u'resource_id': u'hr3hy35y5ht4y54hhthrtg43w',
+          u'resource_title': u'Other raster',
+          u'resource_type': u'RasterResource',
+          u'science_metadata_url': u'http://www.hydroshare.org/hsapi/scimeta/hr3hy35y5ht4y54hhthrtg43w/',
+          u'public': True}
+
+        """
+        url = "{url_base}/sysmeta/{pid}/".format(url_base=self.url_base,
+                                                 pid=pid)
+        r = self._request('GET', url)
+        if r.status_code != 200:
+            if r.status_code == 403:
+                raise HydroShareNotAuthorized(('GET', url))
+            elif r.status_code == 404:
+                raise HydroShareNotFound((pid,))
+            else:
+                raise HydroShareHTTPException((url, 'GET', r.status_code))
+
+        resource = r.json()
+        assert(resource['resource_id'] == pid)
```

Comment:
Note that assert statements are disabled when optimization flags are enabled in python.